### PR TITLE
Add L2C TOW decoding

### PR DESCRIPTION
**Pull request by adel-mamin**
This PR add L2C TOW decoding to compute L2C pseudoranges.
It is contingent on the following PRs:
https://github.com/swift-nav/libswiftnav-private/pull/6
https://github.com/swift-nav/libswiftnav/pull/357
https://github.com/swift-nav/piksi_firmware/pull/672

### CODE REVIEW

File: src/decode/decode_gps_l2c.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  /* Process incoming nav bits */
+  s8 soft_bit;
+  while (tracking_channel_nav_bit_get(channel_info->tracking_channel,
+                                      &soft_bit)) {
+    /* Update TOW */
+    u8 symbol;
+    u32 delay;
+    s32 tow_ms;
+
+    symbol = (soft_bit >= 0) ? MAX_PROBABILITY_OF_SYMBOL_1 :
```
Comment (jacobmcnamee) [inline src/decode/decode_gps_l2c.c]: `symbol = soft_bit + 128` ?

This function
```
/** Compress a 32 bit integration value down to 8 bits.
 *
 * \param bit_integrate   Signed bit integration value.
 */
s8 nav_bit_quantize(s32 bit_integrate)
```
could stand to be improved, but in theory the decoder should be receiving soft values and can hopefully take advantage of them.

File: src/decode/decode_gps_l2c.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+#include <assert.h>
+
+#include "ephemeris.h"
+#include "track.h"
+#include "sbp.h"
+#include "sbp_utils.h"
+#include "signal.h"
+
+/** GPS L2C CNAV message length in symbols */
+#define GPS_CNAV_MSG_LENGTH         (300)
```
Comment (jacobmcnamee) [inline src/decode/decode_gps_l2c.c]: All of these defines should probably be in various places in libswiftnav.

File: src/decode/decode_gps_l2c.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    expected by cnav_msg_decoder_add_symbol(). */
+#define MAX_PROBABILITY_OF_SYMBOL_0        0
+
+/** Maximum probability of symbol 1 as
+    expected by cnav_msg_decoder_add_symbol(). */
+#define MAX_PROBABILITY_OF_SYMBOL_1        0xFF
+
+typedef struct {
+  cnav_msg_t cnav_msg;
+  cnav_msg_decoder_t decoder;
```
Comment (jacobmcnamee) [inline src/decode/decode_gps_l2c.c]: Can we call this `cnav_msg_decoder` so it's a little less confusing? There are already a lot of decoders floating around.

**Comment by jacobmcnamee** [comment]
Added some comments to `Add L2C TOW decoding`. Can we also move the modifications to track.c and track_gps_l2cm.c into the L2 tracking PR?

File: src/decode/decode_gps_l2c.c (hunk 1 above)
Comment (adel-mamin) [inline src/decode/decode_gps_l2c.c]: You are right. My original approach was too harsh. I have changed the code. It should be fixed now.

File: src/decode/decode_gps_l2c.c (hunk 2 above)
Comment (adel-mamin) [inline src/decode/decode_gps_l2c.c]: Should be fixed now.

File: src/decode/decode_gps_l2c.c (hunk 3 above)
Comment (adel-mamin) [inline src/decode/decode_gps_l2c.c]: Should be fixed now.

**Comment by adel-mamin** [comment]
I moved the modifications in track.c and track_gps_l2cm.c into L2C tracking PR https://github.com/swift-nav/piksi_firmware/pull/672

**Comment by rge-exafore** [comment]
Moved to private (https://github.com/swift-nav/piksi_firmware_private/pull/9) and closed.

**Pull request closed without merging**